app/models/vm_or_template/operations.rb

```diff
@@ -112,11 +112,12 @@ def validate_vm_control
             end
       unsupported_reason_add(:control, msg) if msg
     end
-  end
 
-  def validate_terminate
-    return {:available => false, :message => 'The VM is terminated'} if self.terminated?
-    {:available => true, :message => nil}
+    supports :terminate do
+      msg = validate_vm_control
+      msg ||= [false, _("Provider doesn't support vm_destroy")] if !ext_management_system.respond_to?(:vm_destroy)
```

Comment:
yes, :+1: for checking the ems method. OTOH I'm thinking if would be good to move this also to the `supports` layer and therefor closer to the ems.

This means, have this check inside BaseManager. This would also guide future provider implementations what it needs to support this. 
What do you think?